packages/tools/src/tools/annotation/UltrasoundDirectionalTool.ts

```diff
@@ -0,0 +1,916 @@
+    if (data.handles.points.length !== 2) {
+      return;
+    }
+
+    const { cachedStats } = data;
+    const targetIds = Object.keys(cachedStats);
+
+    for (let i = 0; i < targetIds.length; i++) {
+      const targetId = targetIds[i];
+
+      const image = this.getTargetIdImage(targetId, renderingEngine);
+
+      // If image does not exists for the targetId, skip. This can be due
+      // to various reasons such as if the target was a volumeViewport, and
+      // the volumeViewport has been decached in the meantime.
+      if (!image) {
+        continue;
+      }
+
+      const { imageData } = image;
+
+      const worldPos1 = data.handles.points[0];
+      const worldPos2 = data.handles.points[1];
+
+      const imageIndex1 = transformWorldToIndex(imageData, worldPos1);
+      const imageIndex2 = transformWorldToIndex(imageData, worldPos2);
+
+      const { values: values1, units: units1 } =
+        getCalibratedProbeUnitsAndValue(image, [imageIndex1]);
+      const { values: values2, units: units2 } =
+        getCalibratedProbeUnitsAndValue(image, [imageIndex2]);
+
+      let xValues, yValues, units, isHorizontal;
+      let isUnitless = false;
+      if (
+        units1[0] !== units2[0] ||
+        units1[1] !== units2[1] ||
+        (units1[0] === 'raw' && units2[0] === 'raw')
+      ) {
+        // if units are not the same, we cannot calculate the diff
+        // so we just report the px distance
+        const value = distanceToPoint(worldPos1, worldPos2);
+
+        xValues = [value, 0];
+        yValues = [value, 0];
+        units = ['px'];
+        isUnitless = true;
+      } else {
+        const canvasPoint1 = enabledElement.viewport.worldToCanvas(worldPos1);
+        const canvasPoint2 = enabledElement.viewport.worldToCanvas(worldPos2);
+
+        const canvasDeltaY = canvasPoint2[1] - canvasPoint1[1];
+        const canvasDeltaX = canvasPoint2[0] - canvasPoint1[0];
+
+        isHorizontal = Math.abs(canvasDeltaX) > Math.abs(canvasDeltaY);
+        xValues = [values1[0], values2[0]];
+        yValues = [values1[1], values2[1]];
+
+        units = [units1[0], units1[1]];
+      }
+
+      cachedStats[targetId] = {
+        xValues,
+        yValues,
+        isHorizontal,
+        units,
+        isUnitless,
+      };
+    }
+
+    annotation.invalidated = false;
+
+    // Dispatching annotation modified
+    const eventType = Events.ANNOTATION_MODIFIED;
+
+    const eventDetail: AnnotationModifiedEventDetail = {
+      annotation,
+      viewportId,
+      renderingEngineId,
+    };
+    triggerEvent(eventTarget, eventType, eventDetail);
+
+    return cachedStats;
+  }
+}
+
+function defaultGetTextLines(data, targetId, configuration): string[] {
+  const cachedStats = data.cachedStats[targetId];
+  const { xValues, yValues, units, isUnitless, isHorizontal } = cachedStats;
+
+  if (isUnitless) {
+    return [`${roundNumber(xValues[0])} px`];
+  }
+
+  if (configuration.displayBothAxesDistances) {
```

Comment:
Thank you - I know this is a bit of a pain, but sometimes it really helps to have both.